src/modules/todolist.js

```diff
@@ -0,0 +1,42 @@
+import Task from './task.js';
+
+class ToDoList {
+  constructor() {
+    this.list = [];
+  }
+
+  addNewTask = (index = null, description = null, isCompleted = null) => {
+    if (description) {
+      const newTaskIndex = index || this.list.length + 1;
+      const newTaskIsCompleted = isCompleted || false;
+      const newTask = new Task(newTaskIndex, description, newTaskIsCompleted);
+      this.list.push(newTask);
+      return newTask;
+    }
+    return null;
+  }
+
+  updateTask = (index, isCompleted) => {
+    this.list[index - 1].isCompleted = isCompleted;
+  }
+
+  updatedTaskDescription = (index, description) => {
+    this.list[index - 1].description = description;
+  }
+
+  removeTask = (todo) => {
+    this.list = this.list.filter((item) => item !== todo);
+    this.list.forEach((task, indexOfTask) => {
+      task.index = indexOfTask + 1;
+    });
+  }
+
+  removeCompletedTask = () => {
+    this.list = this.list.filter((item) => item.isCompleted !== true);
+    this.list.forEach((task, indexOfTask) => {
+      task.index = indexOfTask + 1;
+    });
+  }
+}
+
```

Comment:
- [ ] kindly consider including a function that iterates over the tasks array and populates the HTML list item element for each task.
hint 👇 

```
const populateTasks = (task, parentElement) => {
  const { description, completed, index: id } = task;
  const innerHTML = `
         <p>${description}</p>
         <p>${completed ? 'completed' : 'not completed'}</p>`;
  const newTodoListElement = document.createElement('li');
  Object.assign(newTodoListElement, { innerHTML, id });
  parentElement.appendChild(newTodoListElement);
};
```
After doing so consider the below project requirement for your project to meet the expected output
 - On page load, the task array is rendered dynamically to the DOM. The tasks should appear in order of the `index` values for each task.

Reply:
THis is already working @divinecharlotte

Reply:
please recheck it all the requirement are fulfilled

Reply:
DId this project 2 times already 3rd day project is approved